R/sigmine.R

```diff
@@ -1,88 +1,126 @@
 #' @title load_sigmine
-#' 
+#'
 #' @description Download and filter data on the mines being explored legally in Brazil
-#' 
+#'
 #' @param space_aggregation A string that indicates the level of aggregation of the data. It can be by municipality or state
 #' @param source A string indicating where the raw data is in your computer. The default is NULL which means the data will be extracted directly from the website
-#' 
+#' @param language A string indicating which language is the desired, 'eng' for english and 'pt' for portuguese
+#'
 #' @return A data base with data containing the area of each selected type of soil covering in each selected year
 #'
 #' @importFrom utils unzip
-#' 
+#'
 #' @author DataZoom, Department of Economics, Pontifical Catholic University of Rio de Janeiro
 #'
 #' @encoding UTF-8
 #' @export
 #'
 #' @examples
 #' \dontrun{
-#' load_sigmine(space_aggregation = 'municipality', source = NULL)
+#' load_sigmine(space_aggregation = 'municipality', source = NULL, language = 'pt')
 #' }
 
 
 
 
-load_sigmine<-function(space_aggregation = c('municipality', 'state','municipio','estado'), source = NULL){
+load_sigmine<-function(space_aggregation = c('municipality', 'state','municipio','estado'), source = NULL, language = c('eng', 'pt')){
   if(is.null(source)){
-    url1<-'http://sigmine.dnpm.gov.br/sirgas2000/brasil.zip'
     p1f <- tempfile(fileext = ".zip")
-    download.file(url1, p1f, mode="wb")
+    tryCatch(
+    expr = {
+      url1<-'https://app.anm.gov.br/dadosabertos/SIGMINE/PROCESSOS_MINERARIOS/BRASIL.zip'
+      download.file(url1, p1f, mode="wb")
+    },
+    error = function(e){
+      url1<-'http://sigmine.dnpm.gov.br/sirgas2000/brasil.zip'
+      download.file(url1, p1f, mode="wb")
+    }
```

Comment:
Acho que vale a pena escrever uma mensagem para aparecer no console do R acusando qual a fonte os dados estao sendo baixados. E alertar, se um site nao funcionar, que esta sendo baixado de outro site. Isso pode inclusve nos ajudar a saber de qual fonte de dados estamos gerando os dados.  

Ou seja, algo do tipo:

`
warning(Website DMPN failed. Proceeding with the data from ANM)
`

Reply:
O comando já faz isso, pelo menos quando eu rodei apareceu algo similar (aparecia o primeiro site e se não conseguisse baixar aparecia que não foi possível baixar e seguia para o outro, também mostrando de qual site estava baixando).